syncstorage-spanner/src/stream.rs

```diff
@@ -144,19 +144,19 @@ where
     }
 
     // We could implement Stream::poll_next instead of this, but
-    // this is easier for now and we can refactor into the trait later
-    pub async fn next_async(&mut self) -> Option<DbResult<Vec<Value>>> {
+    // this is easier for now and we can refactor into the trait later.
+    pub async fn next_async(&mut self) -> DbResult<Option<Vec<Value>>> {
         while self.rows.is_empty() {
             match self.consume_next().await {
-                Ok(true) => (),
-                Ok(false) => return None,
+                Ok(true) => {}
+                Ok(false) => return Ok(None),
                 // Note: Iteration may continue after an error. We may want to
                 // stop afterwards instead for safety sake (it's not really
                 // recoverable)
-                Err(e) => return Some(Err(e)),
+                Err(e) => return Err(e),
```

Comment:
We can kill this `Err` arm of the match by propagating it earlier via `self.consume_next().await?` -- and once you do that, you'll notice the match statement itself is no longer necessary (and killing it is a nice cleanup)